test/register-bson.js

```diff
@@ -8,10 +8,31 @@
 
 // This should be done by mocha --require, but that isn't supported until mocha version 7+
 require('chai/register-expect');
+require('array-includes/auto');
+require('object.entries/auto');
 
 const BSON = require('../lib/bson');
 const { ensureBuffer } = require('../lib/ensure_buffer');
-
 BSON.ensureBuffer = ensureBuffer;
 
+const { Assertion, util } = require('chai');
+Assertion.overwriteMethod('throw', function (original) {
+  return function assertThrow(...args) {
+    if (args.length === 0 || args.includes(BSON.BSONError) || args.includes(BSON.BSONTypeError)) {
+      // By default, lets check for BSONError or BSONTypeError
+      // Since we compile to es5 instanceof is broken???
+      const assertion = original.apply(this, args);
+      const object = util.flag(assertion, 'object');
+      return this.assert(
+        object && /BSONError|BSONTypeError/.test(object.stack),
+        'expected #{this} to be a BSONError or a BSONTypeError but got #{act}',
+        'expected #{this} to not be a BSONError nor a BSONTypeError but got #{act}',
+        (object && object.stack) || '<error undefined>'
+      );
```

Comment:
If someone makes use of a plain Error going forward this is what the test would fail with.
```
AssertionError: expected [Error: bson size must be >= 5, is 1] to throw a BSONError or a BSONTypeError but got [Error: bson size must be >= 5, is 1]
```

Reply:
Can you remind me why we need to do the method overrides or why we needed to create the new error classes to begin with? My memory of last week's discussions is a bit hazy...

Reply:
We covered this offline, basically have to do some extra leg work because of compilation level and capturing intentional lib errors as opposed to misuse of API.